### PR TITLE
Basic PHP support (AST edges etc.) and DeclarationUsage edge providing

### PR DESCRIPTION
**Pull request by max-martynov**

### CODE REVIEW

**Comment by egor-bogomolov** [comment]
Begins implementation of #32

File: psiminer-core/src/main/kotlin/psi/method/PHPMethodProvider.kt
Comment (egor-bogomolov) [inline psiminer-core/src/main/kotlin/psi/method/PHPMethodProvider.kt:16]: Error message should mention body. Let's move the exceptions to a separate file in order to avoid duplicating and keeping them consistent across languages.

File: psiminer-core/src/main/kotlin/psi/method/PHPMethodProvider.kt
Comment (egor-bogomolov) [inline psiminer-core/src/main/kotlin/psi/method/PHPMethodProvider.kt:19]: It would be cleaner to use the `PhpDocComment` interface rather than its implementation, as in the future there might be a different implementation of this interface.

File: psiminer-core/src/main/kotlin/psi/method/PHPMethodProvider.kt
Comment (egor-bogomolov) [inline psiminer-core/src/main/kotlin/psi/method/PHPMethodProvider.kt:23]: Same about `PhpDocComment`

File: psiminer-core/src/main/kotlin/psi/method/PHPMethodProvider.kt
Comment (egor-bogomolov) [inline psiminer-core/src/main/kotlin/psi/method/PHPMethodProvider.kt:43]: Can you cast to `PsiMethod` or `MethodImpl` is strictly necessary here?

File: psiminer-core/src/main/kotlin/psi/transformations/excludenode/ExcludePhpSymbolsTransformation.kt
Comment (egor-bogomolov) [inline psiminer-core/src/main/kotlin/psi/transformations/excludenode/ExcludePhpSymbolsTransformation.kt:8]: If you want to use `in` it might be a good idea to make a set rather than a list.

**Review by egor-bogomolov — COMMENTED** [review]
Awesome job 🚀  Could you please fix a few minor comments before merging?

File: build.gradle.kts
Comment (max-martynov) [reply build.gradle.kts:60]: I moved the declaration of this plugin to the gradle.properties, however the version specification is still required. So it builds only when like this:

kotlin.code.style=official

platformType=IU
platformVersion=222.4345.14
platformDownloadSources=true
platformPlugins = com.intellij.java, org.jetbrains.kotlin, android, maven, gradle, Groovy, com.jetbrains.php:222.4345.14
...

Is it okay?

File: psiminer-cli/src/main/kotlin/config/StorageConfigs.kt
Comment (max-martynov) [reply psiminer-cli/src/main/kotlin/config/StorageConfigs.kt:53]: Oops 🤪
Wanted to change it but forgot...

File: psiminer-core/src/main/kotlin/psi/method/PHPMethodProvider.kt
Comment (max-martynov) [reply psiminer-core/src/main/kotlin/psi/method/PHPMethodProvider.kt:43]: I tested casting to `PsiMethod` and it didn't work :(

File: psiminer-core/src/main/kotlin/Common.kt
Comment (max-martynov) [reply psiminer-core/src/main/kotlin/Common.kt:28]: Well, hard to say because I didn't manage to find any examples of such files. Even the oldest PHP files on GitHub have `.php` extension so I'd rather not support any other formats. If you have another opinion, we can always change it quickly.

File: psiminer-core/src/main/kotlin/psi/method/PHPMethodProvider.kt
Comment (egor-bogomolov) [reply psiminer-core/src/main/kotlin/psi/method/PHPMethodProvider.kt:43]: Weird, let's put here a comment with TODO to investigate it later

File: psiminer-core/src/main/kotlin/psi/method/MethodProvider.kt
Comment (egor-bogomolov) [inline psiminer-core/src/main/kotlin/psi/method/MethodProvider.kt:24]: Let's change wording here. I suggest something like this: "Cannot extract $absentToken : class $className is not a method". For this, you should also pass the encountered class name.

File: gradle.properties
Comment (egor-bogomolov) [inline gradle.properties:6]: I did not know it, but `gradle.properties` actually [does not support string interpolation](https://stackoverflow.com/questions/23853044/does-gradle-support-properties-interpolation) 😞 

Let's keep it like this for now, but I guess we should move constants to build file or some separate file in the future.

**Review by egor-bogomolov — APPROVED** [review]
Great, now the PR looks good!

**Pull request merged**
